fsm/fsm.go

```diff
@@ -320,3 +320,16 @@ func NewErrWaitingForStateTimeout(expected,
 		expected, actual,
 	))
 }
+
+type ErrWaitingForStateEarlyAbortError error
```

Comment:
nit: godoc

Reply:
Also consider the usual pattern where we create a new error global with `errors.New` and then when you add context you just `fmt.Errorf("blah %w blah ...", myError, ...)`